compiler/include/symbol.h

```diff
@@ -361,6 +379,45 @@ class ArgSymbol : public LcnSymbol {
 *                                                                   *
 ********************************* | ********************************/
 
+class ShadowVarSymbol : public VarSymbol {
+public:
+  ShadowVarSymbol(ForallIntentTag iIntent, const char* iName, Expr* iSpec = NULL);
+
+  virtual void    verify();
+  virtual void    accept(AstVisitor* visitor);
+  DECLARE_SYMBOL_COPY(ShadowVarSymbol);
+
+  virtual void    replaceChild(BaseAST* oldAst, BaseAST* newAst);
+  virtual bool    isConstant()                              const;
+  virtual bool    isConstValWillNotChange()                 const;
+
+  const char* intentDescrString() const;
+  // The corresponding outer var or NULL if not applicable.
+  SymExpr* outerVarSE() const { return (SymExpr*)outerVarRep; }
+  bool     isReduce()   const { return intent == TFI_REDUCE;  }
+  Expr*    spec()       const;
```

Comment:
Need a comment saying what this spec function does. It's not immediately obvious to me.

Reply:
I will add this:
// The reduce intent specification.

Reply:
I'm still confused about this. What is a reduce intent specification? Why is it an Expr? Can you give an example?

Reply:
Given this code:
```chpl
var x: MyReduceResultType;
forall ... with (MyReduceOp(int) reduce x) { ... }
```
spec() is the CallExpr for `MyReduceOp(int)`.

This btw means that I want to reduce using MyReduceOp class and the type of the input data is `int`.

Reply:
So maybe we should call the method something like `reduceOpExpr` ?

Reply:
`spec` just doesn't communicate it to me...

Reply:
reduceOpExpr() it is.